File: backend/app/interface/api/v1/project.py
from typing import List
from fastapi import APIRouter, Depends, HTTPException
from app.application.project.service import ProjectApplicationService
from app.application.project.dto import (
    ProjectCreateDTO,
    ProjectDTO,
    ServerCreateDTO,
    ServerDTO,
    ServerMetricsDTO
)
from .dependencies import get_project_service
from app.application.tenant.service import get_current_tenant

router = APIRouter(prefix="/projects", tags=["projects"])

@router.post("", response_model=ProjectDTO)
async def create_project(
    dto: ProjectCreateDTO,
    current_tenant = Depends(get_current_tenant),
    service: ProjectApplicationService = Depends(get_project_service)
) -> ProjectDTO:
    """创建项目"""
    return await service.create_project(tenant_id=current_tenant.id, dto=dto)

@router.post("/{project_id}/servers", response_model=ServerDTO)
async def add_server(
    project_id: str,
    dto: ServerCreateDTO,
    service: ProjectApplicationService = Depends(get_project_service)
) -> ServerDTO:
    """添加服务器"""
    try:
        return await service.add_server(project_id, dto)
    except ValueError as e:
        raise HTTPException(status_code=404, detail=str(e))

@router.put("/servers/{server_id}/metrics")
async def update_server_metrics(
    server_id: str,
    dto: ServerMetricsDTO,
    service: ProjectApplicationService = Depends(get_project_service)
) -> None:
    """更新服务器指标"""
    try:
        await service.update_server_metrics(server_id, dto)
    except ValueError as e:
        raise HTTPException(status_code=404, detail=str(e)) 

File: frontend/src/stores/project.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { http } from '@/utils/http'

export const useProjectStore = defineStore('project', () => {
  const currentTenant = ref(null)
  const projects = ref([])

  // 获取项目列表
  const fetchProjects = async () => {
    const response = await http.get(`/tenants/${currentTenant.value.id}/projects`)
    projects.value = response.data
    return response.data
  }

  // 创建项目
  const createProject = async (data) => {
    const response = await http.post('/projects', data)
    return response.data
  }

  // 获取项目详情
  const fetchProjectDetail = async (projectId: string) => {
    const response = await http.get(`/projects/${projectId}`)
    return response.data
  }

  // 获取项目服务器列表
  const fetchProjectServers = async (projectId: string) => {
    const response = await http.get(`/projects/${projectId}/servers`)
    return response.data
  }

  // 添加服务器
  const addServer = async (projectId: string, data) => {
    const response = await http.post(`/projects/${projectId}/servers`, data)
    return response.data
  }

  // 移除服务器
  const removeServer = async (projectId: string, serverId: string) => {
    await http.delete(`/projects/${projectId}/servers/${serverId}`)
  }

  // 更新服务器指标
  const updateServerMetrics = async (serverId: string, data) => {
    await http.put(`/projects/servers/${serverId}/metrics`, data)
  }

  return {
    projects,
    fetchProjects,
    createProject,
    fetchProjectDetail,
    fetchProjectServers,
    addServer,
    removeServer,
    updateServerMetrics
  }
}) 